### DISCUSSION TITLE
How to enable Subsonic API using ReverseProxyAuthentication (was: Header Auth Disable Auto Registration)

### DISCUSSION THREAD

**Discussion by HalianElf** [navidrome/navidrome#2237] [Q&A]
So #[2087](https://github.com/navidrome/navidrome/pull/2087) made it so it auto registers users now. Can there please be a setting to disable this? 

I had it set up so there was no auth on it so the apps could work which worked great since when the username didn't match, it prompted for login. Now I get a Guest account created because the header passes that when it's not a logged in user.

**Comment by deluan** [comment]
In #2087 we assumed that when setting up the Reverse Proxy config, the proxy would not allow "Guest", unauthenticated access to the service. Is this your case? Is your reverse proxy configured to allow unauthenticated access?

**Reply by HalianElf** [reply]
Yes, otherwise D-Sub and other apps wouldn't be able to use the API. I'm trying to play around with my nginx config so I can get both to work but so far it seems to be one or the other - I can have SSO when I log into my dashboard and end up with a Guest account or I remove the header auth and have double logins for the webui so that my apps work.

**Reply by deluan** [reply]
I don't think introducing a config for this is the way to go. The best way to handle this is to allow unauthenticated traffic to the Subsonic API and Public endpoints: `/rest` and `/share`. This way, you will always have to login to access Navidrome's web UI, but Subsonic clients will work just fine.

I moved this discussion to the Q&A category, and maybe others can help you with that.

**Reply by kgarner7** [reply]
Are you using nginx auth request? If so, you can just disable that on /rest and /share.

**Reply by HalianElf** [reply]
I figured out a way to make it work. Just removing auth from the `/rest` and `/api` endpoints made the header auth not work at all. That was basically the first thing I had tried.
